src/LiveComponent/tests/Fixture/Component/Component6.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Tests\Fixture\Component;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\RangeType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\ComponentWithFormTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+/**
+ * @author Jakub Caban <kuba.iluvatar@gmail.com>
+ */
+#[AsLiveComponent('component6')]
+class Component6 extends AbstractController
+{
+    use ComponentWithFormTrait;
+    use DefaultActionTrait;
+
+    private FormBuilderInterface $builder;
+
+    public function __construct(FormBuilderInterface $builder)
+    {
+        $this->builder = $builder;
+    }
+
+    protected function instantiateForm(): FormInterface
+    {
+        $this->builder
+            ->add('text', TextType::class)
+            ->add('textarea', TextareaType::class)
+            ->add('range', RangeType::class)
+            ->add('choice', ChoiceType::class, [
+                'choices' => [
+                    'foo' => 1,
+                    'bar' => 2,
+                ],
+            ])
+            ->add('choice_expanded', ChoiceType::class, [
+                'choices' => [
+                    'foo' => 1,
+                    'bar' => 2,
+                ],
+                'expanded' => true,
+            ])
```

Comment:
What about a `ChoiceType` with `expanded` true AND `multiple` true (i.e. checkboxes)? Is that correctly handled? Or does the check for `$child->vars['expanded']` mess up that case? I'm genuinely just not sure - I'd love it if you could verify by testing that in a real project.

Reply:
Thanks for quick review! I've followed your suggestions in other places and allowed myself to resolve them.

You are probably right about `multiple` option - I have missed that. But when experimenting with component with form I've found that `CheckboxType` was also misbehaving (haven't got enough time yet to investigate it or write a test case for bug report), therefore my intuition tells me that `expanded` and `multiple` (that expands into multiple checkboxes) isn't working right now anyways.

My intent is to spend my OSS Friday next week to investigate problems with checkboxes and either fix them or provide meaningful bug report. I can do that as part of this PR or the next one if you prefer to have at least fix for `expanded` merged earlier :)

Reply:
Thank you! 

> My intent is to spend my OSS Friday next week to investigate problems with checkboxes and either fix them or provide meaningful bug report.

This sounds wonderful. Tbh, I believe I wrote this original code that has the bug(s) in it. I knew I needed to handle the checkbox case, but I'm not certain that I dove deeply (or thought deeply enough) about all of these `expanded` and `multiple` situations. Someone needs to dive deep and look at the whole problem. If you can do that, it would be *wonderful*.

> I can do that as part of this PR or the next one if you prefer to have at least fix for expanded merged earlier :)

Let's get everything fixed up on this PR :) 

Thanks!

Reply:
I have updated PR with fixes for single checkboxes (their `null` value was getting lost after any update) and also `expanded` and `multiple` choice fields.

As it is right now all basic data types (except `FileType`, which is another story) that I've tested are working as intended.

To check out how checkbox inputs are working I used https://github.com/weaverryan/live-demo - as it is right now writing anything in registration form in e-mail first and then clicking checkbox throws in JS. With this PR it synchronizes data between frontend and backend.

To see `expanded`+`multiple` in action you can use the same registration form from my fork: https://github.com/Lustmored/live-demo